### ISSUE TITLE
Enable UltraSSD when deploying VM that supports it

### ISSUE THREAD

**Issue by rverma-dev** [Azure/karpenter-provider-azure#119]
### Tell us about your request

Trying to enabled and use ultra-ssd for nodepool created by karpenter

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

converting below to Nodepool

```
az aks nodepool add --name tikv \
    --cluster-name  \
    --resource-group ${resourceGroup} \
    --node-vm-size ${nodeType} \
    --zones 1 2 3 \
    --aks-custom-headers EnableAzureDiskFileCSIDriver=true \
    --node-count 3 \
    --labels dedicated=tikv \
    --node-taints dedicated=tikv:NoSchedule \
    --enable-ultra-ssd
```

### Are you currently working around this issue?

No 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by Bryce-Soghigian** [comment]
Thanks for creating an issue! 

Leaving some notes as Ultra SSD and my inital thoughts on how it conflicts with karpenter today(not to say we will never implement)

1. Ultra disks can't be used as OS disks, they can only be created as empty data disks. Currently we only are creating the os disk in current karpenter implementation. 
2. [Ultra Disks require zonal knowledge to know if its available](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-ultra-ssd?tabs=azure-portal#vms-using-availability-zones).  This introduces a new dimension to karpenter's requirements

We need to design and think through data disks and requirements being zonal aware.

**Comment by rverma-dev** [comment]
Yeah the usecase is not for OS disks for sure. However I believe to start with we need to find a way to modify AdditionalCapabilities.UltraSSDEnabled = true.

**Comment by tallaxes** [comment]
We should add enabling UltraSSD by default when using VM SKU that supports it, in supported zone - we have this information from SKU API.

**Comment by PabloTriv** [comment]
Merged PR
